WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorView.swift

```diff
@@ -301,27 +302,28 @@ private extension ProductSelectorView {
                 ServiceLocator.analytics.track(event: .ProductListFilter.productListViewFilterOptionsTapped(source: source.filterAnalyticsSource))
             }
             .buttonStyle(LinkButtonStyle())
-            .fixedSize()
         }
-        .padding(.horizontal, insets: safeAreaInsets)
         .background(Color(.listForeground(modal: false)))
     }
 
     @ViewBuilder private var productSelectorHeaderSearchRow: some View {
-        HStack {
-            SearchHeader(text: $viewModel.searchTerm, placeholder: Localization.searchPlaceholder, onEditingChanged: { isEditing in
-                searchHeaderisBeingEdited = isEditing
-            })
-            .padding(.horizontal, insets: safeAreaInsets)
-            .accessibilityIdentifier("product-selector-search-bar")
-            Picker(selection: $viewModel.productSearchFilter, label: EmptyView()) {
-                ForEach(ProductSearchFilter.allCases, id: \.self) { option in Text(option.title) }
+        GeometryReader { geometry in
+            HStack {
+                SearchHeader(text: $viewModel.searchTerm, placeholder: Localization.searchPlaceholder, onEditingChanged: { isEditing in
+                    searchHeaderisBeingEdited = isEditing
+                })
+                .accessibilityIdentifier("product-selector-search-bar")
+                Picker(selection: $viewModel.productSearchFilter, label: EmptyView()) {
+                    ForEach(ProductSearchFilter.allCases, id: \.self) { option in Text(option.title) }
+                }
+                .if(horizontalSizeClass == .regular && geometry.size.width < Constants.headerSearchRowWidth) { $0.pickerStyle(.menu) }
+                .if(horizontalSizeClass == .compact || geometry.size.width > Constants.headerSearchRowWidth) { $0.pickerStyle(.segmented) }
+                .padding(.trailing)
+                .renderedIf(searchHeaderisBeingEdited)
             }
-            .pickerStyle(.segmented)
-            .padding(.leading)
-            .padding(.trailing)
-            .renderedIf(searchHeaderisBeingEdited)
         }
+        .frame(height: Constants.minimumRowHeight * scale)
```

Comment:
Why do we need to set the height?

Reply:
Good question, it's because of the `GeometryReader`. If we do not enforce a desired height for the view, it will take all the available space vertically, so needs to be constrained. If we don't, depending on the view, it would look like this:

![Simulator Screenshot - iPhone 15 Plus - 2024-02-23 at 11 06 50](https://github.com/woocommerce/woocommerce-ios/assets/3812076/51249cf2-a798-4cbc-91b5-9b952d91d100)

I've documented this on 78e717fdc29cb65dddfc31d52359a0bb385e668a

Reply:
It's possible to put the geometry reader on the background of the views you care about, which prevents it messing with their layout. It always feels a little clunky, but perhaps it would be the right choice here to avoid setting the height ourselves...

I would rather find a design which avoids the need for a geometry reader though, if possible.

Reply:
TIL about using the geometry reader on the background. I'll experiment a combination of this and creating a separate component as you advised (sort of AdaptiveStack) separately and see if we can remove the geometry reader.